actor/src/main/scala/org/apache/pekko/pattern/Patterns.scala

```diff
@@ -126,7 +126,8 @@ object Patterns {
    *   timeout);
    * }}}
    */
-  def askWithReplyTo(actor: ActorRef, messageFactory: japi.Function[ActorRef, Any], timeout: Timeout): Future[AnyRef] =
+  def askWithReplyTo(actor: ActorRef, messageFactory: japi.function.Function[ActorRef, Any], timeout: Timeout)
```

Comment:
I am not aboard with this. This is a breaking change where we did not deprecate the old code first.
I don't know what is the best way to approach this but I think this change needs serious debate.

Would it be feasible to have 2 copies of this method? This version and also the old version but that is marked as deprecated.

Reply:
That will not work when using java lambda

Reply:
what would the migration instructions be for affected users?

Reply:
the binary compat issues are large:
https://github.com/apache/pekko/actions/runs/16704071539/job/47279560921?pr=2001

Reply:
Yes, this pain is one time; otherwise, we will have to live with two functions forever ...

Reply:
> the binary compat issues are large: https://github.com/apache/pekko/actions/runs/16704071539/job/47279560921?pr=2001

We do not care about binary compatibility for 2.0.x, so this is a non issue.

There is a fair point about these functions not being deprecated, I am on board with marking the `org.apache.pekko.japi.*` functions as deprecated for the 1.2.x series of Pekko (which is still yet to have its final release) and then this PR to me is approvable.

The whole point of 2.0.x is to remove cruft like this, we are allowed to break binary compatibility. Semver allows this in an epoch version change

Reply:
@mdedetrich So you mean we depracate this classes in 1.2.x right? thanks.